packages/automated-dispute/tests/services/protocolProvider.spec.ts

```diff
@@ -739,4 +741,290 @@ describe("ProtocolProvider", () => {
+                    ],
+                    data: "0x0000000000000000000000000000000000000000000000000000000000000003",
+                    blockNumber: 1n,
+                    transactionHash:
+                        "0x1234567890123456789012345678901234567890123456789012345678901234",
+                    transactionIndex: 1,
+                    blockHash: "0x1234567890123456789012345678901234567890123456789012345678901234",
+                    logIndex: 1,
+                    removed: false,
+                },
+            ];
+
+            (protocolProvider["readClient"] as any).getLogs = vi.fn().mockResolvedValue(mockLogs);
+
+            vi.spyOn(protocolProvider as any, "decodeLogData").mockReturnValue({
+                requestId: "0x123",
+                epoch: 1n,
+                chainId: "eip155:1",
+                request: {},
+            });
+
+            const result = await (protocolProvider as any).getEBORequestCreatorEvents(0n, 100n);
+
+            expect(result).toHaveLength(1);
+            expect(result[0]).toEqual({
+                name: "RequestCreated",
+                blockNumber: 1n,
+                logIndex: 1,
+                rawLog: mockLogs[0],
+                requestId: "0x123",
+                metadata: {
+                    epoch: 1n,
+                    chainId: "eip155:1",
+                    request: {},
+                    requestId: "0x123",
+                },
+            });
+        });
+    });
+
+    describe("getEvents", () => {
+        it("successfully merges and sorts events from all sources", async () => {
+            const protocolProvider = new ProtocolProvider(
+                mockRpcConfig,
+                mockContractAddress,
+                mockedPrivateKey,
+            );
+
+            const mockRequestCreatorEvents = [
+                { name: "RequestCreated", blockNumber: 1n, logIndex: 0 },
+                { name: "RequestCreated", blockNumber: 3n, logIndex: 0 },
+            ];
+
+            const mockOracleEvents = [
+                { name: "ResponseProposed", blockNumber: 2n, logIndex: 0 },
+                { name: "ResponseDisputed", blockNumber: 2n, logIndex: 1 },
+            ];
+
+            vi.spyOn(protocolProvider as any, "getEBORequestCreatorEvents").mockResolvedValue(
+                mockRequestCreatorEvents,
+            );
+            vi.spyOn(protocolProvider as any, "getOracleEvents").mockResolvedValue(
+                mockOracleEvents,
+            );
+
+            const result = await protocolProvider.getEvents(0n, 100n);
+
+            expect(result).toEqual([
+                { name: "RequestCreated", blockNumber: 3n, logIndex: 0 },
+                { name: "ResponseDisputed", blockNumber: 2n, logIndex: 1 },
+                { name: "ResponseProposed", blockNumber: 2n, logIndex: 0 },
+                { name: "RequestCreated", blockNumber: 1n, logIndex: 0 },
+            ]);
```

Comment:
If this test case is passing tests, we need to invert the `mergeEventStreams` old sorting logic to have this pass with blocks sorted the other way round.
```suggestion
            expect(result).toEqual([
                { name: "RequestCreated", blockNumber: 1n, logIndex: 0 },
                { name: "ResponseDisputed", blockNumber: 2n, logIndex: 0 },
                { name: "ResponseProposed", blockNumber: 2n, logIndex: 1 },
                { name: "RequestCreated", blockNumber: 3n, logIndex: 0 },
            ]);
```